wrapper/nodejs/packages/atom-parsetools/astgen.js

```diff
@@ -154,25 +152,60 @@ const fileToJsAst = (file, projectType) => {
   if (file.endsWith(".vue") || file.endsWith(".svelte")) {
     return toVueAst(file);
   }
-  return codeToJsAst(readFileSync(file, "utf-8"), projectType);
+  if (file.endsWith(".ejs")) {
+    return toEjsAst(file);
+  }
+  return codeToJsAst(file, readFileSync(file, "utf-8"), projectType);
 };
 
 /**
  * Convert a single JS/TS code snippet to AST
  */
-const codeToJsAst = (code, projectType) => {
-  const optionsToUse =
-    projectType === "flow" ? babelFlowParserOptions : babelParserOptions;
+const codeToJsAst = (file, code, projectType) => {
+  const isJs = /\.(js|jsx|cjs|mjs)$/.test(file);
+  if (isJs && projectType === "flow") {
+    try {
+      return parseHermes(code, {
+        sourceType: "unambiguous",
+        babel: true,
+        allowReturnOutsideFunction: true,
+        flow: "all",
+        sourceFilename: file,
+        tokens: true
+      });
+    } catch (err) {
+      // Ignore
+    }
+  }
+  // If user explicitly said 'flow', we try Babel-Flow first, then Babel-Standard.
+  // Otherwise, we try Babel-Standard (TS/ESNext) first, then Babel-Flow and finally hermes.
+  let primaryBabelOptions = babelParserOptions;
+  let secondaryBabelOptions = babelFlowParserOptions;
+  if (projectType === "flow") {
+    primaryBabelOptions = babelFlowParserOptions;
+    secondaryBabelOptions = babelParserOptions;
+  }
   try {
-    return parse(code, optionsToUse);
-  } catch {
+    return parse(code, primaryBabelOptions);
```

Comment:
How many parsers do you need to support javascript? The true answer could be infinity, but currently we use three parsers.